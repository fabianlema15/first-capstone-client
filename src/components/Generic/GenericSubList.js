import React from 'react';
import GenericContext from '../../contexts/GenericContext'
import GenericSubItem from './GenericSubItem';
import Helper from '../Utils/Helper'

class GenericList extends React.Component{
  static contextType = GenericContext;

  renderObjs() {
    const { objSubList = [] } = this.context
    return Object.keys(objSubList).map(key =>
      <GenericSubItem key={objSubList[key].id} obj={objSubList[key]} objLabel={Helper.serializeObj(this.context.getSubObjArray)}/>
    )
  }

  render(){
    return <div>
      {Object.keys(this.context.objSubList).length>0 ? this.renderObjs() : 'You still have no products added'}
    </div>
  }
}

export default GenericList;
