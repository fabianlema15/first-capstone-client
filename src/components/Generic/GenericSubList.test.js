import React from 'react';
import ReactDOM from 'react-dom';
import GenericSubList from './GenericSubList'
import toJson from 'enzyme-to-json'
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('<GenericSubList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <GenericSubList>
      </GenericSubList>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Renders GenericSubList as expected', () => {
    const context = { objName: 'Test', objSubList: {}};
	  const wrapper = shallow(<GenericSubList/>, { context });
	  expect(toJson(wrapper)).toMatchSnapshot();
	  });

})
