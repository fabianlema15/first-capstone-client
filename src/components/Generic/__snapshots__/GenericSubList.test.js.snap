// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenericSubList /> Renders GenericSubList as expected 1`] = `
<div>
  You still have no products added
</div>
`;
